homeassistant/components/remote/harmony.py

```diff
@@ -44,29 +45,62 @@
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Harmony platform."""
     import pyharmony
-    global DEVICES
 
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    port = config.get(CONF_PORT)
-    _LOGGER.debug("Loading Harmony platform: %s", name)
-
-    harmony_conf_file = hass.config.path(
-        '{}{}{}'.format('harmony_', slugify(name), '.conf'))
+    host = None
+    activity = None
+
+    if CONF_DEVICE_CACHE not in hass.data:
+        hass.data[CONF_DEVICE_CACHE] = []
+
+    if discovery_info:
+        override = next(c for c in hass.data[CONF_DEVICE_CACHE]
```

Comment:
This will raise a `StopIteration` if nothing matches. If you want it to return `None` or `False`, pass that as a second argument to `next`.

![image](https://user-images.githubusercontent.com/1444314/27065606-274a79f6-4fb3-11e7-8d15-40181ce82089.png)

![image](https://user-images.githubusercontent.com/1444314/27065615-3cf16670-4fb3-11e7-8a03-0927dbd9e869.png)